packages/wrangler/src/__tests__/core/command-registration.test.ts

```diff
@@ -0,0 +1,477 @@
+import { normalizeOutput } from "../../../e2e/helpers/normalize";
+import {
+	COMMAND_DEFINITIONS,
+	defineAlias,
+	defineCommand,
+	defineNamespace,
+} from "../../core/define-command";
+import { mockConsoleMethods } from "../helpers/mock-console";
+import { runInTempDir } from "../helpers/run-in-tmp";
+import { runWrangler } from "../helpers/run-wrangler";
+
+describe("Command Registration", () => {
+	runInTempDir();
+	const std = mockConsoleMethods();
+
+	beforeEach(() => {
+		COMMAND_DEFINITIONS.length = 0; // clears commands definitions so tests do not conflict with eachother
```

Comment:
Would it be better to provide a util to clears commands?

Reply:
No it's only necessary for testing itself. No other test file should need to do this.